clustering/marshalling/protostream/pom.xml

```diff
@@ -45,11 +45,31 @@
         Also contains wildfly-clustering-marshalling-spi implementations using ProtoStream.
     </description>
 
-    <properties>
-        <jdk.max.version>9</jdk.max.version>
-        <!-- This property otherwise causes the java8-test execution to crash -->
-        <modular.jdk.args></modular.jdk.args>
-    </properties>
+    <profiles>
+        <profile>
+            <id>disable-tests-on-publish-to-maven-central</id>
+            <activation>
+                <!--
+                     Disabling tests execution during deployment to the maven central repository.
```

Comment:
This discussion of deployment isn't quite right. I don't think it's needed and it confused me as to what the use case was. Something more along the lines of what you're saying in comments to Paul seems better.

We don't use any SE other than 8 when deploying.  That's long been managed via human policy, but with https://github.com/wildfly/wildfly/pull/14317 it is now enforced.

We use -DskipTests when deploying. We test before doing the maven execution that deploys.

Also, we deploy to JBoss Nexus, not maven central.

I think it's better to talk about the need for this for development workflows, not release tasks.

Reply:
This change is needed in case one tries to build WildFly with JDK16+ @bstansberry .

Reply:
I tried to find better solution but after 2 hours of experimenting with it I gave it up and came with this compromise.